From f25626a68a2fb83b02ee7a13dd4887d1305bc068 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tiziano=20M=C3=BCller?= <tiziano.mueller@stepping-stone.ch>
Date: Tue, 20 May 2014 18:48:05 +0200
Subject: [PATCH 1/2] Introduce --enable/disable-webdav option

This makes the phodav dependency configureable.
And name it after the corresponding channel.
---
 configure.ac | 21 +++++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index d89bd6f..70ab75c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -270,10 +270,23 @@ AC_SUBST(GTHREAD_LIBS)
 PKG_CHECK_MODULES(SOUP, libsoup-2.4)
 AC_SUBST(SOUP_CFLAGS)
 AC_SUBST(SOUP_LIBS)
+AC_ARG_ENABLE([webdav],
+  AS_HELP_STRING([--enable-webdav=@<:@auto/yes/no@:>@],
+                 [Enable webdav support @<:@default=auto@:>@]),
+  [],
+  [enable_webdav="auto"])
+
+if test "x$enable_webdav" = "xno"; then
+  have_phodav="no"
+else
+  PKG_CHECK_MODULES(PHODAV, [libphodav-1.0], [have_phodav=yes], [have_phodav=no])
+  AC_SUBST(PHODAV_CFLAGS)
+  AC_SUBST(PHODAV_LIBS)
 
-PKG_CHECK_MODULES(PHODAV, [libphodav-1.0], [have_phodav=yes], [have_phodav=no])
-AC_SUBST(PHODAV_CFLAGS)
-AC_SUBST(PHODAV_LIBS)
+  if test "x$have_phodav" = "xno" && test "x$enable_webdav" = "xyes"; then
+    AC_MSG_ERROR([webdav support explicitly requested, but some required packages are not available])
+  fi
+fi
 AS_IF([test "x$have_phodav" = "xyes"],
        AC_DEFINE(USE_PHODAV, [1], [Define if supporting phodav]))
 
@@ -754,7 +767,7 @@ AC_MSG_NOTICE([
         Smartcard support:        ${have_smartcard}
         USB redirection support:  ${have_usbredir} ${with_usbredir_hotplug}
         DBus:                     ${have_dbus}
-        PhoDAV:                   ${have_phodav}
+        WebDAV support:           ${have_phodav}
 
         Now type 'make' to build $PACKAGE
 
-- 
1.9.2

